### ISSUE TITLE
While working with storage policies, I noticed hardcoding the UUID in the policy worked but matching the auth.uid() with the user ID always fails.

### ISSUE THREAD

**Issue by mypalvikram** [supabase-community/supabase-swift#227]
# Bug report

I'm working on creating storage policies that allow users to insert, update, and delete items if their user ID is the same as the folder name. Most of this was given from the storage policy templates. The issue is this doesn't work. If I hard-code the UUID, it will work. Example:

Insert, update, and delete policy. This fails.
```
((bucket_id = 'avatars'::text) AND ((auth.uid())::text = (storage.foldername(name))[1]))

Returns:
StorageError(statusCode: "403", message: "new row violates row-level security policy", error: "Unauthorized")
```

Now if I update the policy to a hardcoded UUID like so, it works:
```
(bucket_id = 'avatars'::text) AND ( '5714F7D7-108B-4D9C-8945-8707075845B0'::text) = ((storage.foldername(name))[1])
```

This is the function that I'm using to handle my avatar uploads:
```
    func uploadUserAvatar(image: PhotosPickerItem) async throws -> () {
        guard let imageData = try await image.loadTransferable(type: Data.self) else { return }
        
        let userId = try await supabaseClient.auth.session.user.id
        
        print(userId.uuidString)
                
        let filePath = "\(userId.uuidString)/avatar.jpeg"
        
        try await supabaseClient.storage
            .from("avatars")
            .upload(
                path: filePath,
                file: imageData,
                options: FileOptions(contentType: "image/jpeg", upsert: false)
            )
    }
```

To summarize, even though I've verified that the UUID is the matching UUID on the auth table. When working with storage policies it seems like there is something wrong when `auth.uid()` is being checked on the database storage side or what is being sent back from the session.

**Comment by mypalvikram** [comment]
I did some testing by creating a React Native application with the JavaScript Client and the same storage policies. Everything works as expected.

**Comment by grdsdev** [comment]
Hi @mypalvikram thanks for opening this issue, I'll investigate.

**Comment by grdsdev** [comment]
Hey @mypalvikram I did a few tests and I think I found the issue, I hope it is the same for your case.

Try using the following RLS policy:

```sql
create policy "Give users access to own folder" on storage.objects
  for select to authenticated
    using (bucket_id = 'avatars'
      and lower(auth.uid()::text) = lower((storage.foldername(name))[1]));
```

Notice it is applying the `lower` function to the user id, which is needed because Swift UUID is uppercased, and Postgres is lowercased, so you need to standardize it.

Or, you should lowercase the user id when creating the file path, as:

```swift
let filePath = "\(userId.uuidString.lowercased())/avatar.jpeg"
```

Can you try one of those and let me know if it fixes your issue?
